Sources/StreamChat/APIClient/Endpoints/Payloads/IdentifiablePayload.swift

```diff
@@ -0,0 +1,224 @@
+extension ChannelMemberListPayload: IdentifiablePayloadProxy {
+    func fillIds(cache: inout [String: Set<String>]) {
+        members.fillIds(cache: &cache)
+    }
+}
+
+extension ChannelListPayload: IdentifiablePayloadProxy {
+    func fillIds(cache: inout [String: Set<String>]) {
+        channels.fillIds(cache: &cache)
+    }
+}
+
+extension ChannelPayload: IdentifiablePayloadProxy {
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+        channel.fillIds(cache: &cache)
+        watchers?.fillIds(cache: &cache)
+        membership?.fillIds(cache: &cache)
+        messages.fillIds(cache: &cache)
+        pinnedMessages.fillIds(cache: &cache)
+        channelReads.fillIds(cache: &cache)
+    }
+}
+
+extension ChannelDetailPayload: IdentifiablePayload {
+    var databaseId: String? { cid.rawValue }
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = ChannelDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+        createdBy?.fillIds(cache: &cache)
+        members?.fillIds(cache: &cache)
+        invitedMembers.fillIds(cache: &cache)
+    }
+}
+
+extension UserPayload: IdentifiablePayload {
+    var databaseId: String? { id }
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = UserDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+    }
+}
+
+extension MessagePayload: IdentifiablePayload {
+    var databaseId: String? { id }
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = MessageDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+        user.fillIds(cache: &cache)
+        quotedMessage?.fillIds(cache: &cache)
+        mentionedUsers.fillIds(cache: &cache)
+        threadParticipants.fillIds(cache: &cache)
+        latestReactions.fillIds(cache: &cache)
+        ownReactions.fillIds(cache: &cache)
+        pinnedBy?.fillIds(cache: &cache)
+        pinnedBy?.fillIds(cache: &cache)
+    }
+}
+
+extension MessageReactionPayload: IdentifiablePayload {
+    var databaseId: String? {
+        MessageReactionDTO.createId(userId: user.id, messageId: messageId, type: type)
+    }
+
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = MessageReactionDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+        user.fillIds(cache: &cache)
+    }
+}
+
+extension MemberPayload: IdentifiablePayload {
+    var databaseId: String? { nil } // Cannot build id without channel id
+    static let modelClass: (NSManagedObject & IdentifiableModel).Type? = MemberDTO.self
+
+    func fillIds(cache: inout [String: Set<String>]) {
+        addId(cache: &cache)
+        user.fillIds(cache: &cache)
+    }
+}
+
+extension ChannelReadPayload: IdentifiablePayload {
+    var databaseId: String? { nil } // Needs a composed predicate 'channel.cid == %@ && user.id == %@'
```

Comment:
Shouldn't we add `id` to DTO then? Like `AttachmentId` or `MessageReactionId`

Reply:
We should, yeah, but outside of the scope of this PR for now